src/main/java/org/opensearch/jdbc/types/TypeConverters.java

```diff
@@ -56,12 +57,37 @@ public class TypeConverters {
         tcMap.put(JDBCType.BINARY, new BinaryTypeConverter());
 
         tcMap.put(JDBCType.NULL, new NullTypeConverter());
+
+        // Adding Struct Support
+        tcMap.put(JDBCType.STRUCT, new StructTypeConverter());
     }
 
     public static TypeConverter getInstance(JDBCType jdbcType) {
         return tcMap.get(jdbcType);
     }
 
+    public static class StructTypeConverter extends BaseTypeConverter {
+
+        private static final Set<Class> supportedJavaClasses = Collections.unmodifiableSet(
+                new HashSet<>(Arrays.asList(
+                        Struct.class
+                )));
```

Comment:
Maybe `... = Set.of(Struct.class);`? If it is available in this version of java.

Reply:
Yeah so for where we are using the JDBC driver we need java 8 support, so wouldn't be able to use the java 11 collections apis.